app/controllers/tutees_controller.rb

```diff
@@ -32,20 +23,26 @@ def edit
 
     #Update all of the attributes gathered from edit form
     def update
-        @time_availabilitys_ids = params[:tutee][:time_availabilitys_ids]
-        @tutee = Tutee.find(params[:id])
-        @tutee.update_attributes!(tutee_params)
-        @tutee.time_availabilitys.destroy_all
-        if @time_availabilitys_ids != nil
-            @time_availabilitys_ids.each do |id|
-                time = TimeAvailability.find(id)
-                @tutee.time_availabilitys << time
+        begin
+            @time_availabilitys_ids = params[:tutee][:time_availabilitys_ids]
+            @tutee = Tutee.find(params[:id])
+            @tutee.update_attributes!(tutee_params)
+            @tutee.time_availabilitys.destroy_all
+            if @time_availabilitys_ids != nil
+                @time_availabilitys_ids.each do |id|
+                    time = TimeAvailability.find(id)
+                    @tutee.time_availabilitys << time
+                end
             end
-        end
-        flash[:notice] = "Form for #{@tutee.first_name + ' ' + @tutee.last_name} was succesfully created"
-        redirect_to tutor_match_path(@tutee)
+            flash[:notice] = "Form for #{@tutee.first_name + ' ' + @tutee.last_name} was succesfully created"
+            redirect_to tutor_match_path(@tutee)
+        rescue ActiveRecord::RecordInvalid => invalid   
+            flash[:error] = invalid.record.errors
+            redirect_to_correct_edit_form(@tutee, :tutee)
```

Comment:
I think you can just do 'redirect_to edit_tutee_path(@tutee)' here without the custom method